client/jetpack-cloud/sections/partner-portal/utils.ts

```diff
@@ -201,3 +201,18 @@ export function isJetpackBundle( product: APIProductFamilyProduct | string ) {
 	}
 	return product.family_slug === 'jetpack-packs';
 }
+
+/**
+ *	Checks if the license keys are compatible with multisite.
+ *
+ * @param licenseKeys
+ * @returns boolean indicating if the license keys are compatible with multisite
+ */
+export function isMultisiteCompatibleLicenseKeys( licenseKeys: string[] ): boolean {
+	const INCOMPATIBLE_LICENSE_KEY_PREFIXES = [ 'jetpack-backup', 'jetpack-scan' ];
+
+	const isIncompatibleLicenseKey = ( licenseKey: string ) =>
+		!! INCOMPATIBLE_LICENSE_KEY_PREFIXES.find( ( prefix ) => licenseKey?.startsWith( prefix ) );
+
+	return ! licenseKeys.find( ( licenseKey ) => isIncompatibleLicenseKey( licenseKey ) );
```

Comment:
I think it will be faster and more readable if we use [`.some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) or [`.every()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) here instead of [`.find()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find), because `.some()` and `.every()` return early if the condition matches. Also, we won't have to use double negation (`!!`) for a boolean value.

So, we can re-write it like this:

```ts
/**
 * Whether the license keys are compatible with WP Multisite.
 *
 * @param {Array<string>} licenseKeys
 * @returns {boolean} indicating if the license keys are compatible with multisite
 */
export function areLicenseKeysMultisiteCompatible( licenseKeys: Array< string > ): boolean {
	const INCOMPATIBLE_LICENSE_KEY_PREFIXES = [ 'jetpack-backup', 'jetpack-scan' ];

	return licenseKeys.every( ( licenseKey ) => {
		return INCOMPATIBLE_LICENSE_KEY_PREFIXES.every(
			( prefix ) => ! licenseKey?.startsWith( prefix )
		);
	} );
}
```

Or, if you like Regex, we can make it a one-liner

```ts
/**
 * Whether the license keys are compatible with WP Multisite.
 *
 * @param {Array<string>} licenseKeys
 * @returns {boolean} indicating if the license keys are compatible with multisite
 */
export function areLicenseKeysMultisiteCompatible( licenseKeys: Array< string > ): boolean {
	return licenseKeys.every( ( key ) => ! /^jetpack-(backup|scan)/.test( key ) );
}
```

Reply:
IMHO, the regex one is more readable from left to right:

`.every` licence `key` should pass the `.test` i.e. it does not `!` start with `^` `jetpack-backup` or `jetpack-scan`

Reply:
@manzoorwanijk I initially was planning to use regex but thought the list might grow in the future, but I guess it makes sense, for now, to use it for readability. 👍 thanks for the suggestion.